src/Symfony/Component/VarDumper/Tests/Caster/MemcachedCasterTest.php

```diff
@@ -0,0 +1,152 @@
+ */
+class MemcachedCasterTest extends TestCase
+{
+    use VarDumperTestTrait;
+
+    public function testCastMemcachedWithDefaultOptions()
+    {
+        if (!class_exists('Memcached')) {
+            $this->markTestSkipped('Memcached not available');
+        }
+
+        $var = new \Memcached();
+        $var->addServer('127.0.0.1', 11211);
+        $var->addServer('127.0.0.2', 11212);
+
+        $expected = <<<EOTXT
+Memcached {
+  servers: array:2 [
+    0 => array:3 [
+      "host" => "127.0.0.1"
+      "port" => 11211
+      "type" => "TCP"
+    ]
+    1 => array:3 [
+      "host" => "127.0.0.2"
+      "port" => 11212
+      "type" => "TCP"
+    ]
+  ]
+  options: {}
+}
+EOTXT;
+        $this->assertDumpEquals($expected, $var);
+    }
+
+    public function testCastMemcachedWithCustomOptions()
+    {
+        if (!class_exists('Memcached')) {
+            $this->markTestSkipped('Memcached not available');
+        }
+
+        $var = new \Memcached();
+        $var->addServer('127.0.0.1', 11211);
+        $var->addServer('127.0.0.2', 11212);
+        $var->setOptions($this->getCustomOptions());
+
+        $expected = <<<'EOTXT'
+Memcached {
+  servers: array:2 [
+    0 => array:3 [
+      "host" => "127.0.0.1"
+      "port" => 11211
+      "type" => "TCP"
+    ]
+    1 => array:3 [
+      "host" => "127.0.0.2"
+      "port" => 11212
+      "type" => "TCP"
+    ]
+  ]
+  options: {
+    OPT_COMPRESSION: %s
+    OPT_PREFIX_KEY: "%s"
+    OPT_USER_FLAGS: %i
+    OPT_STORE_RETRY_COUNT: %i
+    OPT_HASH: %i
+    OPT_DISTRIBUTION: %i
+    OPT_LIBKETAMA_COMPATIBLE: %i
+    OPT_LIBKETAMA_HASH: %i
+    OPT_TCP_KEEPALIVE: %i
+    OPT_BINARY_PROTOCOL: %i
+    OPT_NO_BLOCK: %i
+    OPT_TCP_NODELAY: %i
+    OPT_SOCKET_SEND_SIZE: %i
+    OPT_SOCKET_RECV_SIZE: %i
+    OPT_CONNECT_TIMEOUT: %i
+    OPT_RETRY_TIMEOUT: %i
+    OPT_DEAD_TIMEOUT: %i
+    OPT_SEND_TIMEOUT: %i
+    OPT_RECV_TIMEOUT: %i
+    OPT_POLL_TIMEOUT: %i
+    OPT_SERVER_FAILURE_LIMIT: %i
+    OPT_AUTO_EJECT_HOSTS: %i
+    OPT_HASH_WITH_PREFIX_KEY: %i
+    OPT_NOREPLY: %i
+    OPT_SORT_HOSTS: %i
+    OPT_USE_UDP: %i
+    OPT_NUMBER_OF_REPLICAS: %i
+    OPT_RANDOMIZE_REPLICA_READ: %i
+    OPT_REMOVE_FAILED_SERVERS: %i
+    OPT_SERVER_TIMEOUT_LIMIT: %i
+  }
+}
+EOTXT;
+
+        $this->assertDumpMatchesFormat($expected, $var);
+    }
+
+    private function getCustomOptions()
+    {
+        $optionsToIgnore = array(
+            \Memcached::OPT_SERIALIZER, // trying to set this option to something else then default throws Memcached::setOptions(): invalid serializer provided
+            \Memcached::OPT_CACHE_LOOKUPS, // option is deprecated
+            \Memcached::OPT_VERIFY_KEY, // trying to set this option throws Memcached::setOption(): error setting memcached option: INVALID ARGUMENTS
+        );
+
+        $optionConstants = MemcachedCaster::getMemcachedOptionConstants();
+        $defaultOptions = MemcachedCaster::discoverDefaultMemcachedOptions();
```

Comment:
Using these methods in tests in nos enough a reason to make them public. We should to otherwise, like hardcoding a few more things - not testing all options but just a subset is fine.